### PR TITLE
Wrapper function

### PR DESCRIPTION
**Pull request by salmanap**
Adding a wrapper function that is able to call any one of the supported financial functions. This way customers can get all the supported financial from a single lamnda

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: financial_functions/lambda_handlers.py [hunk 1]
```diff
@@ -334,3 +333,23 @@ def sln_handler(request, context):
...
+def financial_function_generic_handler(request, context):
+    """
+    This function takes in an arbritary financial function and its parameters as inputs and returns the result of that calculation
+    :param request: Dict containing the financial function name and its parameters
+    :param context: Lambda execution context
+    :return: Dict with a 'result' entry containing the result of the calculation
+    """
+    logger.info("financial function request: {}".format(request))
+
+    function_name = request.get("functionName")
```
Comment (jlhood) [inline financial_functions/lambda_handlers.py]: We've been using underscores for request params, so let's be consistent and change to `function_name`. Also, I think this is more pythonic:

```
if 'function_name' not in request:
    return {'error': '...'}
```

File: financial_functions/lambda_handlers.py [hunk 2]
```diff
@@ -334,3 +333,23 @@ def sln_handler(request, context):
...
+    if function_name is None: 
+        return {"error": "Please provide a function name using the functionName parameter"}
+        
+    function_handler_name = function_name + "_handler"
+    if hasattr(sys.modules[__name__], function_handler_name):
```
Comment (jlhood) [inline financial_functions/lambda_handlers.py]: Be careful here. If someone passes `financial_function_generic` as the handler name you'll recursively call yourself to death. I was picturing putting this generic handler in a separate file/module and checking the module lambda_handlers for the corresponding handler function. I was thinking the other file would be called api_handler.py and would be expecting an API GW lambda proxy request, but I understand that's a TODO item for this. I would suggest moving to a diff file, but keeping the generic lambda so you're not wrestling with the API GW integration.

File: financial_functions/lambda_handlers.py [hunk 3]
```diff
@@ -334,3 +333,23 @@ def sln_handler(request, context):
     args = [request['cost'], request['salvage'], request['life']]
     return __call_ff('sln', args)
 
+def financial_function_generic_handler(request, context):
```
Comment (jlhood) [inline financial_functions/lambda_handlers.py]: Can you just call this wrapper_handler?

File: templates/generic.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: 'AWS::Serverless-2016-10-31'
+
+Resources:
+  # Wrapper function for all financial functions
+  EFFECT:
```
Comment (jlhood) [inline templates/generic.yaml]: WRAPPER

File: test/unit/test_lambda_handlers.py [hunk 5]
```diff
@@ -8,6 +8,26 @@
...
+        "rate": 0.004166666666667,
+        "nper": 120,
+        "pmt": -100
+    }
+    
+    response = handlers.financial_function_generic_handler(request, None)
+    assert 'result' in response
+    assert round(response.get('result'), 6) == 15528.227945
+    
+    #test function does not
```
Comment (jlhood) [inline test/unit/test_lambda_handlers.py]: PUt this in a separate test method

File: test/unit/test_lambda_handlers.py [hunk 6]
```diff
@@ -8,6 +8,26 @@
...
+    response = handlers.financial_function_generic_handler({"functionName":"not_available"}, None)
+    assert 'error' in response
+    
+    #test functionName parameter passed in is incorrect
```
Comment (jlhood) [inline test/unit/test_lambda_handlers.py]: Put this in a separate test method

File: Makefile [hunk 7]
```diff
@@ -129,5 +129,11 @@ set-sln-template:
 package-sln: set-sln-template package
 deploy-sln: set-sln-template package-sln deploy
 
+set-generic-template:
```
Comment (jlhood) [inline Makefile]: Change "generic" to "wrapper" in all these lines

**Comment by salmanap** [comment]
updated code based on past review

**Pull request merged**